core/eds.go

```diff
@@ -19,22 +17,12 @@ import (
 // extendBlock extends the given block data, returning the resulting
 // ExtendedDataSquare (EDS). If there are no transactions in the block,
 // nil is returned in place of the eds.
-func extendBlock(data types.Data) (*rsmt2d.ExtendedDataSquare, error) {
-	if len(data.Txs) == 0 && data.SquareSize == uint64(1) {
+func extendBlock(data types.Data, appVersion uint64) (*rsmt2d.ExtendedDataSquare, error) {
+	if app.IsEmptyBlock(data, appVersion) {
 		return nil, nil
 	}
 
-	sqr, err := square.Construct(data.Txs.ToSliceOfBytes(), appconsts.LatestVersion, share.MaxSquareSize)
-	if err != nil {
-		return nil, err
-	}
```

Comment:
unrelated to this release, fwiw, node was already constructing the square in the correct way. I was originally correct when saying that rc7 was not breaking to celestia-node (notice the usage of `share.MaxSquareSize`)